src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -93,7 +93,7 @@ public bool TryAdd(Document file, PublishItem item)
 
                 foreach (var conflictingFile in conflictingFiles)
                 {
-                    if (_publishItems.TryRemove(conflictingFile, out var item) && item.Path != null)
+                    if (_publishItems.TryRemove(conflictingFile, out var item) && item.Path != null && (conflictingFile.ContentType != ContentType.Resource || conflictingFile.Docset.Config.Output.CopyResources))
```

Comment:
@928PJY , can you change the test framework to assert that tests shouldn't change any files in source folder? You can simplify compare the last modified time of each file in the source repo.